include/zephyr/devicetree/gpio.h

```diff
@@ -230,6 +230,180 @@ extern "C" {
 #define DT_GPIO_FLAGS(node_id, gpio_pha) \
 	DT_GPIO_FLAGS_BY_IDX(node_id, gpio_pha, 0)
 
+/**
+ * @brief Get the number of GPIO hogs in a node
+ *
+ * This expands to the number of hogged GPIOs, or zero if there are none.
+ *
+ * Example devicetree fragment:
+ *
+ *     gpio1: gpio@... {
+ *       compatible = "vnd,gpio";
+ *       #gpio-cells = <2>;
+ *
+ *       n1: node-1 {
+ *               gpio-hog;
+ *               gpios = <0 GPIO_ACTIVE_HIGH>, <1 GPIO_ACTIVE_LOW>;
+ *               output-high;
+ *       };
+ *
+ *       n2: node-2 {
+ *               gpio-hog;
+ *               gpios = <3 GPIO_ACTIVE_HIGH>;
+ *               output-low;
+ *       };
+ *     };
+ *
+ * Bindings fragment for the vnd,gpio compatible:
+ *
+ *     gpio-cells:
+ *       - pin
+ *       - flags
+ *
+ * Example usage:
+ *
+ *     DT_NUM_GPIO_HOGS(DT_NODELABEL(n1)) // 2
+ *     DT_NUM_GPIO_HOGS(DT_NODELABEL(n2)) // 1
+ *
+ * @param node_id node identifier; may or may not be a GPIO hog node.
+ * @return number of hogged GPIOs in the node
+ */
+#define DT_NUM_GPIO_HOGS(node_id) \
+	COND_CODE_1(IS_ENABLED(DT_CAT(node_id, _GPIO_HOGS_EXISTS)), \
+		    (DT_CAT(node_id, _GPIO_HOGS_NUM)), (0))
+/**
+ * @brief Get the node identifier for the controller phandle from a
+ *        GPIO hog node at an index
+ *
+ * Example devicetree fragment:
+ *
+ *     gpio1: gpio@... {
+ *       compatible = "vnd,gpio";
+ *       #gpio-cells = <2>;
+ *
+ *       n1: node-1 {
+ *               gpio-hog;
+ *               gpios = <0 GPIO_ACTIVE_HIGH>, <1 GPIO_ACTIVE_LOW>;
+ *               output-high;
+ *       };
+ *
+ *       n2: node-2 {
+ *               gpio-hog;
+ *               gpios = <3 GPIO_ACTIVE_HIGH>;
+ *               output-low;
+ *       };
+ *     };
+ *
+ * Bindings fragment for the vnd,gpio compatible:
+ *
+ *     gpio-cells:
+ *       - pin
+ *       - flags
+ *
+ * Example usage:
+ *
+ *     DT_GPIO_HOG_CTLR_BY_IDX(DT_NODELABEL(n1), 0) // DT_NODELABEL(gpio1)
+ *     DT_GPIO_HOG_CTLR_BY_IDX(DT_NODELABEL(n1), 1) // DT_NODELABEL(gpio1)
+ *     DT_GPIO_HOG_CTLR_BY_IDX(DT_NODELABEL(n2), 0) // DT_NODELABEL(gpio1)
```

Comment:
Minor: the example might be improved if you had a hog with node label `n3` under a different GPIO controller with label `gpio2` as well, to show the parent/child relationship more clearly.

Reply:
You are right. I considered this, but left it out since I didn't want to clutter the documentation too much. There's already a lot of macros to document for each `gpiohogs-macro` addition. but if you'd like, I can extend all the examples with a third GPIO hog node under a second GPIO controller.

Reply:
That's OK; keeping things short is valuable too. Thanks for explaining.